src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -403,30 +559,103 @@ public SymbolicMatch FindMatch(RegexRunnerMode mode, ReadOnlySpan<char> input, i
             }
             else
             {
-                Registers endRegisters = _pattern._info.ContainsLineAnchor ?
+                Registers endRegisters = _containsAnyAnchor ?
                     FindSubcaptures<FullInputReader>(input, matchStart, matchEnd, perThreadData) :
                     FindSubcaptures<NoZAnchorInputReader>(input, matchStart, matchEnd, perThreadData);
                 return new SymbolicMatch(matchStart, matchEnd - matchStart, endRegisters.CaptureStarts, endRegisters.CaptureEnds);
             }
         }
 
+        /// <summary>
+        /// This version of <see cref="FindEndPositionFallback"/> uses a different set of interfaces,
+        /// which don't check for many inner loop edge cases e.g. input end or '\n'.
+        /// All edge cases are handled before entering the loop.
+        /// </summary>
+        private int FindEndPositionOptimized<TOptimizedInputReader, TAcceleratedStateHandler, TOptimizedNullabilityHandler>(
+            ReadOnlySpan<char> input, int pos, long timeoutOccursAt, RegexRunnerMode mode, PerThreadData perThreadData)
+            where TOptimizedInputReader : struct, IOptimizedInputReader
+            where TAcceleratedStateHandler : struct, IAcceleratedStateHandler
+            where TOptimizedNullabilityHandler : struct, IOptimizedNullabilityHandler
+        {
+            // this initial state candidate is not really used in the common DFA case
```

Comment:
When is it used?

Reply:
I'm unsure if this is needed anywhere at all, this was part of the original end position lookup